students/BrianB/lesson03/mailroom1.py

```diff
@@ -1,121 +1,109 @@
-#  donor[1].pop()  Leaving pop() blank, deletes
-#                       the value in the last
-#                       position
-#----------------------------------------------
+# ----------------------------------------------
 
 import sys
 import math
 from textwrap import dedent
 
-print("In the Mailroom")
+#print("In the Mailroom")
 
 # convert lists to a dictionary - see Assignment 05, ToDo.py
 
 # A tuple of donor and listing of past donations
 # You want a tuple to keep a record of past donations. The original
 # tuple stays in as assigned memory location.
 donor_data = [("Guile", [1000, 589, 25000]),
-                ("Blanca", [23, 1, 17]),
-                ("Ken", [1001, 590, 25001]),
-                ("M. Bison", [15000, 8000, 1200000]),
-                ("Chun-Li", [61000000, 500000, 1200000]),
-                ]
+              ("Blanca", [23, 1, 17]),
+              ("Ken", [1001, 590, 25001]),
+              ("M. Bison", [15000, 8000, 1200000]),
+              ("Chun-Li", [61000000, 500000, 1200000]),
+              ]
+
+
+def gen_donor():
+    # print("The Donors Are:")
+    for donor in donor_data:
+        print(donor[0])
 
 
 def thank_you():
     while True:
-        name = input("Which Donor would you like to see?\n\
-    Please enter the full name only.\n\
-    >>> ")
+        name = input("Which Donor would you like to see?\n"
+                     "Please enter the full name only.\n"
+                     ">>> ")
         if name == "list":
-            for donor in donor_data:
-                print(donor[0])
+            gen_donor()
             continue
         for donor in donor_data:
             if name == donor[0]:
-                print("There's a match!\n", donor[0])
-                new_amt = input("Please enter a donation amount:")
-                donor[1].append(float(new_amt))
-                # print(donor_data)
-                print(dedent('''
-
-                            Dear {}!
-                             Thank for your donation of ${:,.2f}! We appreciate
-                             your contribution and without you nothing would
-                             be possible!
-
-                                                     Sincerely,
-                                                        Street Fighter, LLC
-                                                            an equal opportunity
-                                                            employer
-                                '''.format(donor[0], donor[1][-1])))
-                break  # this does not break out of [thank =_you()]
+                print(donor[0], "was found in our database!")
+                new_amt = input("Please enter a new donation amount:")
+                donor[1].append(int(new_amt))
+                print(donor_data)
+                start_over = input("If you would like to try another "
+                                   "donor enter 'yes' or 'no'.")
+                if start_over.lower() == "yes":
```

Comment:
Be careful here because the break here kicks you out of the forloop but not the while loop.
If you enter a new user and type in "no" for the start_over, it still prompts you for another. 

You will want to put a break in the for loop where you find the user and the if start_over logic just outside the for loop